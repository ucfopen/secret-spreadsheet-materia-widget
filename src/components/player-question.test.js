import React from 'react'
import Question from './player-question'
import renderer from 'react-test-renderer';

describe('Question component', () => {
	beforeEach(() => {
		jest.resetModules();
	});

	test('Is rendered with question body', () => {
		const props = {
			handleQuestionToggle: jest.fn(),
			entryQuestion: 'Test entry question',
			questionBody: 'Test question body'
		}

		const component = renderer.create(<Question {... props} />);
		const tree = component.toJSON();

		expect(tree).toMatchSnapshot();
	});

	test('Is rendered without question body', () => {
		const props = {
			handleQuestionToggle: jest.fn(),
			entryQuestion: 'Test entry question',
			questionBody: ''
		}

		const component = renderer.create(<Question {... props} />);
		const tree = component.toJSON();

		expect(tree).toMatchSnapshot();
	});
});
