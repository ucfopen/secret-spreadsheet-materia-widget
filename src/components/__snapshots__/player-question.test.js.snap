// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Question component Is rendered with question body 1`] = `
<div
  className="popup question"
>
  <div>
    <p
      className="mainQuestion"
    >
      Test entry question
    </p>
    <p>
      Test question body
    </p>
    <button
      onClick={[MockFunction]}
      type="button"
      value="Next"
    >
      Next
    </button>
  </div>
</div>
`;

exports[`Question component Is rendered without question body 1`] = `
<div
  className="popup question"
>
  <div>
    <p
      className="mainQuestion"
    >
      Test entry question
    </p>
    <button
      onClick={[MockFunction]}
      type="button"
      value="Next"
    >
      Next
    </button>
  </div>
</div>
`;
